src/api/index.js

```diff
@@ -343,6 +358,82 @@ export function getInstallProperties() {
   return get(uri);
 }
 
+
+export async function patchServiceAccount(
+  serviceAccountName,
+  namespace,
+  secretName
+) {
+  const uri = getKubeAPI('serviceaccounts', {
+    name: serviceAccountName,
+    namespace
+  });
+  const patch1 = await patchAddSecret(uri, secretName);
+  return patch1;
+}
+
+// Get list of service accounts where secrets patched to, returns service account in full
+export async function getSecretServiceAccountList(saList, secretName) {
+  const secretServiceAccountList = [];
+
+  if (saList.length > 0) {
+    saList.forEach(element => {
+      const saSecretList = element.secrets;
+      let secretFound = false;
+      saSecretList.forEach(element1 => {
+        if (element1.name === secretName) {
+          secretFound = true;
+        }
+      });
+      if (secretFound === true) {
+        secretServiceAccountList.push(element);
+      }
+    });
+  }
+
+  return secretServiceAccountList;
+}
+
+export function getIndexOfSecret(sa, secretName) {
```

Comment:
first param here looks like it's an array of secrets, not a service account as the param name would suggest

also why are we using index here, seems odd. Is the order guaranteed? Should we not be using name / uid or something more stable?

if we are sticking with index, [findIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) is a native implementation

Reply:
The order is guaranteed as it does not change unless we add or delete secrets. Because of the way the patch command works in the KubeAPI, it is impossible to remove by name or anything more stable (we have tried this) and therefore we must remove by index. However this works fine as the correct index will always be able to be found.